code/datums/state_machine/state.dm

```diff
@@ -0,0 +1,36 @@
+// An individual state, defined as a `/decl` to save memory.
+// On a directed graph, these would be the nodes themselves, connected to each other by unidirectional arrows.
+/decl/state
+	// Transition decl types, which get turned into refs to those types.
+	// Note that the order DOES matter, as decls earlier in the list have higher priority
+	// if more than one becomes 'open'.
+	var/list/transitions = null
+
+// Called by `SSstate_machines` to have the state decls build the directed graph themselves.
+/decl/state/proc/set_up()
```

Comment:
As discussed on Discord, it may be worth looking into making this an Initialize() override to avoid having to have an init subsystem.